src/main/java/org/jabref/model/search/rules/GrammarBasedSearchRule.java

```diff
@@ -232,7 +232,7 @@ public Boolean visitComparison(SearchParser.ComparisonContext context) {
             if (fieldDescriptor.isPresent()) {
                 return comparison(fieldDescriptor.get().getText(), ComparisonOperator.build(context.operator.getText()), right);
             } else {
-                return new ContainBasedSearchRule(caseSensitive).applyRule(right, entry);
+                return SearchRules.getSearchRule(caseSensitive, regex).applyRule(right, entry);
```

Comment:
@Siedlerchr perhaps the `if` solution is better here? This might be too frivolous application of the DRY principle.

Reply:
Doesn't really matter in this case, think it's okay. Didn't think about this way.